Web UI env var reference (#1640)
temporalio/documentation: docs-src/references/web-ui-environment-variables.md

    -e TEMPORAL_UI_PORT=8080 \
    -e TEMPORAL_AUTH_ENABLED=true \
    -e TEMPORAL_AUTH_PROVIDER_URL=https://accounts.google.com \
    -e TEMPORAL_AUTH_CLIENT_ID=xxxxx-xxxx.apps.googleusercontent.com \
    -e TEMPORAL_AUTH_CLIENT_SECRET=xxxxxxxxxxxxxxx \
    -e TEMPORAL_AUTH_CALLBACK_URL=https://xxxx.com:8080/auth/sso/callback \
    -e TEMPORAL_UI_ENABLED=true \
    -e TEMPORAL_OPENAPI_ENABLED=true \
    -e TEMPORAL_TLS_CA=../ca.cert \
    -e TEMPORAL_TLS_CERT=../cluster.pem \
    -e TEMPORAL_TLS_KEY=../cluster.key \
    -e TEMPORAL_TLS_ENABLE_HOST_VERIFICATION=true \
    -e TEMPORAL_TLS_SERVER_NAME=tls-server \
    temporalio/ui:<tag>
``` -->

The environment variables are defined in the [UI server configuration template file](https://github.com/temporalio/ui-server/blob/main/docker/config_template.yaml) and described in more detail below.

## `TEMPORAL_ADDRESS`

The [Frontend Service](/concepts/what-is-a-frontend-service) address for the Temporal Cluster.
This environmental variable can be set [in the base configuration file](/references/web-ui-configuration#temporalgrpcaddress) using `temporalGrpcAddress`.

This variable is required for setting other environmental variables.

> `TEMPORAL_ADDRESS` is also needed when setting up [Cross-Origin Resource Sharing](/references/web-ui-configuration#cors).

Reviewer: where does this come from? 
```suggestion
```

## `TEMPORAL_UI_PORT`

The port used by the Temporal WebUI Server and the HTTP API.

> Defining this variable allows the Cluster and API to communicate effectively.

Reviewer: not sure what this means

This variable is needed for `TEMPORAL_OPENAPI_ENABLED` and all auth-related settings to work properly.

> `TEMPORAL_UI_ENABLED` requires a valid port number.

Reviewer 2: I think the intent was to create some narrative around which env vars might be working together. I agree this is more narrow and more literal than what I imagined.

> `TEMPORAL_UI_ENABLED` requires a valid port number.

Author: I'll keep a few, but I'll purge the other dependencies

## `TEMPORAL_PUBLIC_PATH`

> The location in the Web UI Server where everything is stored.

Reviewer: that’s a value such as “” or “/custom-path” that will make the UI to be served from from a subpath http://localhost:8080/custom-path

## `TEMPORAL_AUTH_ENABLED`

Enables or disables Temporal authentication and authorization methods.

When enabled, the Temporal Cluster will use the provider information in the [UI configuration file](/references/web-ui-configuration#auth) to verify the identity of users.

All auth-related environmental variables depend on `TEMPORAL_AUTH_ENABLED`.
Disabling the variable will retain given values.

All auth-related variables must be defined when this is enabled.

## `TEMPORAL_AUTH_PROVIDER_URL`

> The URL for Temporal's authentication and authorization OIDC provider.

Reviewer: ```suggestion
The .well-known IDP discovery URL for authentication and authorization.
```

Reviewer: as an example the common terminology  https://help.akana.com/content/current/cm/api_oauth/oauth_discovery/m_oauth_getOpenIdConnectWellknownConfiguration.htm

This can be set as in the UI server configuration with [auth](/references/web-ui-configuration#auth).

The Temporal Cluster can be set up to use your preferred authentication and authorization methods.

This variable is required for configuring auth and its related variables.

## `TEMPORAL_AUTH_ISSUER_URL`

> The URL for the authentication or authorization issuer.

Reviewer: this comment explains a little bit more what this is https://github.com/temporalio/ui-server/blob/c73aac1ed166901625f412ddc3dc4ef7be98c2b9/server/config/config.go#L86

## `TEMPORAL_AUTH_CLIENT_ID`

> The client ID for Temporal authentication and authorization methods.

Reviewer: lets use the common terminology https://help.akana.com/content/current/cm/api_oauth/oauth_discovery/m_oauth_getOpenIdConnectWellknownConfiguration.htm

Authentication and authorization need to identify the correct client.

> Providing a Client ID ensures that auth methods will run smoothly.

Reviewer: it doesnt ensure, just one of the required parameters. auth can still be broken

This variable is needed to [configure auth](/references/web-ui-configuration#auth).
Other auth variables must be defined along with `TEMPORAL_AUTH_CLIENT_ID`.

## `TEMPORAL_AUTH_CLIENT_SECRET`

The client secret used for authentication and authorization.

Client Secrets are used by the oAuth Client for authentication.

This variable is needed to configure auth.
Define all auth-related variables when using this variable.

## `TEMPORAL_AUTH_CALLBACK_URL`

The callback URL used by Temporal for authentication and authorization.

> Callback URLs are invoked by the API after it's finished communicating with the Temporal Cluster.

Reviewer: ```suggestion
Callback URLs are invoked by IDP after user finished has authenticated in IDP.
```

This variable should be defined when [configuring auth](/references/web-ui-configuration#auth).

## `TEMPORAL_UI_ENABLED`

Enables or disables the [browser UI](/references/web-ui-configuration#enableui) for the Temporal Cluster.

Enabling the browser UI allows the Server to be accessed from your web browser.
If disabled, the server cannot be viewed on the web, but the UI server APIs remain available for use.

This variable needs to be set to 'true' in order to set `TEMPORAL_OPENAPI_ENABLED`.

## `TEMPORAL_OPENAPI_ENABLED`

Enables or disables OpenAPI features for the Temporal Web UI.

This can be set initially with the [enableOpenAPI](/references/web-ui-configuration#enableopenapi) UI configuration.
The documentation can be found at `/openapi/` on your Temporal Cluster.

This variable requires `TEMPORAL_UI_ENABLED` to be set to 'true'.

## `TEMPORAL_DEFAULT_NAMESPACE`

> The default [Namespace](/concepts/what-is-a-namespace) that the Web UI loads data for.

Reviewer: ```suggestion
The default [Namespace](/concepts/what-is-a-namespace) that the Web UI opens first
```

## `TEMPORAL_TLS_CERT_DATA`

The data obtained from `TEMPORAL_TLS_CERT_DATA`.

`TEMPORAL_TLS_CERT` must be defined to obtain a value for this variable.

## `TEMPORAL_TLS_KEY_DATA`

The data obtained from `TEMPORAL_TLS_KEY_DATA`.

`TEMPORAL_TLS_KEY` must be defined to use this variable.

## `TEMPORAL_TLS_ENABLE_HOST_VERIFICATION`

Enables or disables [Transport Layer Security (TLS) host verification](/references/web-ui-configuration#tls).

When enabled, TLS checks the Host Server to ensure that files are being sent to and from the correct URL.

This variable is needed when configuring TLS and its associated variables.

## `TEMPORAL_TLS_SERVER_NAME`

The server on which to operate [Transport Layer Security (TLS) protocols](/references/web-ui-configuration#tls).

TLS allows the current server to transmit encrypted files to other URLs without having to reveal itself.
Because of this, TLS operates a go-between server.

This variable is needed to configure TLS and its related environmental variables.

## `TEMPORAL_CODEC_ENDPOINT`

The endpoint for the [Codec Server](/concepts/what-is-a-codec-server), if configured.

## `TEMPORAL_CODEC_PASS_ACCESS_TOKEN`

> The access token needed to transport data through the Codec Server.

Reviewer: ```suggestion
Whether to attach JWT access token as ‘authorization’ header in requests with the Codec Server.
```

Make sure that `TEMPORAL_CODEC_ENDPOINT` is defined to use this variable.

## `TEMPORAL_FORWARD_HEADERS`

> The headers needed to forward payloads through the Codec Server.

Reviewer: https://github.com/temporalio/ui-server/blob/c73aac1ed166901625f412ddc3dc4ef7be98c2b9/server/config/config.go#L53
